core/src/main/java/com/nisovin/magicspells/spells/PlayerMenuSpell.java

```diff
@@ -176,16 +178,16 @@ private void processClickSpell(Subspell subspell, Player caster, Player target,
 		subspell.cast(caster, power);
 	}
 
-	private void open(Player opener, SpellData data) {
+	private void open(Player opener, SpellData data, int page) {
 		List<Player> players = new ArrayList<>(Bukkit.getOnlinePlayers());
 		if (!addOpener) players.remove(opener);
 		if (playerModifiers != null) players.removeIf(player -> !playerModifiers.check(player));
 		if (radius > 0) players.removeIf(player -> opener.getLocation().distance(player.getLocation()) > radius);
 
-		int size = (int) Math.ceil((players.size()+1) / 9.0) * 9;
+		int size = (int) Math.ceil(Math.min((players.size()+1), 54) / 9.0) * 9;
 		Inventory inv = Bukkit.createInventory(opener, size, Component.text(internalName));
 
-		for (int i = 0; i < players.size(); i++) {
+		for (int i = (page * maxPlayersPerPage); i < Math.min(players.size(), (page + 1) * maxPlayersPerPage); i++) {
```

Comment:
As mentioned earlier, `maxPlayersPerPage` doesn't make sense as a static field. Just use the value here.